sdks/python/src/opik/api_objects/dataset/dataset_item.py

```diff
@@ -1,37 +1,46 @@
 from typing import Optional, Dict, Any
 import pydantic
-from .. import constants
+import json
+import hashlib
+from .. import constants, helpers
 
 
 class DatasetItem(pydantic.BaseModel):
-    """A DatasetItem object representing an item in a dataset."""
+    """
+    A DatasetItem object representing an item in a dataset.
+    The format is flexible.
+    """
 
-    model_config = pydantic.ConfigDict(strict=True)
+    model_config = pydantic.ConfigDict(extra="allow", strict=False)
 
-    input: Dict[str, Any]
-    """The input data for the dataset item."""
-
-    expected_output: Optional[Dict[str, Any]] = None
-    """The expected output for the dataset item."""
-
-    metadata: Optional[Dict[str, Any]] = None
-    """Additional metadata associated with the dataset item."""
+    id: pydantic.SkipValidation[str] = pydantic.Field(
+        default_factory=helpers.generate_id
+    )
+    """The unique identifier for this dataset item."""
 
     trace_id: Optional[str] = None
     """The ID of the trace associated with this dataset item."""
 
     span_id: Optional[str] = None
     """The ID of the span associated with this dataset item."""
 
-    id: Optional[str] = None
-    """The unique identifier for this dataset item."""
-
     source: str = constants.DATASET_SOURCE_SDK
     """The source of the dataset item. Defaults to DATASET_SOURCE_SDK."""
 
-    def get_content(self) -> Dict[str, Any]:
-        return {
-            "input": self.input,
-            "expected_output": self.expected_output,
-            "metadata": self.metadata,
-        }
+    def get_content(self, include_id: bool = False) -> Dict[str, Any]:
+        content = {**self.model_extra}
+        if include_id:
+            content["id"] = self.id
+
+        return content
+
+    def content_hash(self) -> str:
```

Comment:
we could convert this method to `__hash__()` magick method, and (probably) cache this value with lru_cache/cached_property to speedup re-syncing process

Reply:
1. Whenever we calculate the hash, it's always done on the fresh `DatasetItem` instance which hash function hasn't been called before, so at the moment we'll not benefit from caching (unless we make an external cash function for memorizing content dictionaries, which I think might be a bit of an overkill here at the moment). But it's a good idea in general, as soon as we have a case of a repeated hash calculation or we see that we have performance issues, we should probably get back to it.
2. I specially didn't name this method `__hash__()` to emphasize that it calculates hash only from the data that we consider a dataset item's content :) (id-like fields are not involved in the calculation, as you can see).